test/unified_memory_allocation/common/pool.hpp

```diff
@@ -79,7 +79,11 @@ struct proxy_pool : public pool_base {
 
         memset(ptr, 0, num * size);
 
-        EXPECT_EQ(ret, UMA_RESULT_SUCCESS);
+        try {
+            EXPECT_EQ(ret, UMA_RESULT_SUCCESS);
```

Comment:
Not the biggest fan of this approach, especially since it's repeated multiple times.
Maybe it would be worth defining something like `EXPECT_EQ_NOEXCEPT` ?

Do these functions need to be `noexcept`?

Reply:
Yes, they need to be noexcept to compile. Added a macro.

Reply:
Do ASSERT_* macros also throw? ASSERT might be a better fit here anywhere.

Also, do we actually run the gtest with throw_on_failure? Can we disable it?